lib/saml2.coffee

```diff
@@ -89,13 +106,25 @@ format_pem = (key, type) ->
 
 # Takes a compressed/base64 enoded @saml_request and @private_key and signs the request using RSA-SHA256. It returns
 # the result as an object containing the query parameters.
-sign_get_request = (saml_request, private_key) ->
-  data = "SAMLRequest=" + encodeURIComponent(saml_request) + "&SigAlg=" + encodeURIComponent('http://www.w3.org/2001/04/xmldsig-more#rsa-sha256')
+sign_get_request = (saml_request, private_key, relay_state, response=false) ->
+  action = if response then "SAMLResponse" else "SAMLRequest"
+  data = "#{action}=" + encodeURIComponent(saml_request)
+  if relay_state
+    data += "&RelayState=" + encodeURIComponent(relay_state)
+  data += "&SigAlg=" + encodeURIComponent('http://www.w3.org/2001/04/xmldsig-more#rsa-sha256')
```

Comment:
No. `qs.stringify` does not have a way (that I know of) of specifying the order of the parameters. The SAML spec defines the order specifically and because the data is signed, the order MUST match the spec.